vaadin-grid-pro-flow/src/main/java/com/vaadin/flow/component/gridpro/GridPro.java

```diff
@@ -415,6 +410,66 @@ public boolean getKeepEditorOpen() {
      *
      * @param <E> the bean type
      */
+    @DomEvent("cell-edit-started")
+    public static class CellEditStartedEvent<E> extends ComponentEvent<GridPro<E>> {
+
+        private E item;
+        private String path;
+
+        /**
+         * Creates a new event using the given source and indicator whether the
+         * event originated from the client side or the server side.
+         *
+         * @param source     the source component
+         * @param fromClient <code>true</code> if the event originated from the client
+         * @param item       the item to be edited, provided in JSON as internally represented in Grid
+         * @param path       item subproperty that was changed
+         */
+        public CellEditStartedEvent(GridPro<E> source, boolean fromClient,
+                                        @EventData("event.detail.item") JsonObject item,
+                                        @EventData("event.detail.path") String path) {
+            super(source, fromClient);
+            this.item = source.getDataCommunicator()
+                    .getKeyMapper().get(item.getString("key"));
+            this.path = path;
+        }
+
+        /**
+         * Gets an instance of edited item.
+         *
+         * @return the instance of edited item
+         */
+        public E getItem() {
+            return item;
+        }
+
+
+        /**
+         * Gets the key of the column where item was edited.
+         *
+         * @return the key of the column
+         */
+        private String getPath() {
```

Comment:
While we are breaking stuff, remove the public access modifiers from `ItemPropertyChangeEvent`'s methods `getPath` and `getSourceItem`. Make them either private or package-protected depending on the needs of the implementation.
One of the methods even returns `JsonObject` which is like poison for old-school Java-only Vaadin users.

Reply:
Changed.